Source/Scene/Expression.js

```diff
@@ -0,0 +1,1886 @@
+        }
+    };
+
+    Node.prototype._evaluateLiteralString = function(frameState, feature) {
+        return this._value;
+    };
+
+    Node.prototype._evaluateVariableString = function(frameState, feature) {
+        var result = this._value;
+        var match = variableRegex.exec(result);
+        while (match !== null) {
+            var placeholder = match[0];
+            var variableName = match[1];
+            var property = feature.getProperty(variableName);
+            if (!defined(property)) {
+                property = '';
+            }
+            result = result.replace(placeholder, property);
+            match = variableRegex.exec(result);
+        }
+        return result;
+    };
+
+    Node.prototype._evaluateVariable = function(frameState, feature) {
+        // evaluates to undefined if the property name is not defined for that feature
+        return feature.getProperty(this._value);
+    };
+
+    function checkFeature (ast) {
+        return (ast._value === 'feature');
+    }
+
+    // PERFORMANCE_IDEA: Determine if parent property needs to be computed before runtime
+    Node.prototype._evaluateMemberDot = function(frameState, feature) {
+        if (checkFeature(this._left)) {
+            return feature.getProperty(this._right.evaluate(frameState, feature));
+        }
+        var property = this._left.evaluate(frameState, feature);
+        if (!defined(property)) {
+            return undefined;
+        }
+
+        var member = this._right.evaluate(frameState, feature);
+        if ((property instanceof Cartesian2) || (property instanceof Cartesian3) || (property instanceof Cartesian4)) {
+            // Vector components may be accessed with .r, .g, .b, .a and implicitly with .x, .y, .z, .w
+            if (member === 'r') {
+                return property.x;
+            } else if (member === 'g') {
+                return property.y;
+            } else if (member === 'b') {
+                return property.z;
+            } else if (member === 'a') {
+                return property.w;
+            }
+        }
+        return property[member];
+    };
+
+    Node.prototype._evaluateMemberBrackets = function(frameState, feature) {
+        if (checkFeature(this._left)) {
+            return feature.getProperty(this._right.evaluate(frameState, feature));
+        }
+        var property = this._left.evaluate(frameState, feature);
+        if (!defined(property)) {
+            return undefined;
+        }
+
+        var member = this._right.evaluate(frameState, feature);
+        if ((property instanceof Cartesian2) || (property instanceof Cartesian3) || (property instanceof Cartesian4)) {
+            // Vector components may be accessed with [0][1][2][3], ['r']['g']['b']['a'] and implicitly with ['x']['y']['z']['w']
+            // For Cartesian2 and Cartesian3 out-of-range components will just return undefined
+            if (member === 0 || member === 'r') {
```

Comment:
I don't think there is anything to change now, but given that `member` can be a number or a string, I don't think this will get optimized well.